### PR TITLE
Social providers login

### PR DESCRIPTION
**Pull request by toha87**
Here is the Social Login part for our mirror
https://localhost:3000/config
gives you 3 options for sign in with Google, Facebook, and Amazon 
I know it uses class components it's just how they made their example.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,11 +18,13 @@
...
     "react-open-weather": "^0.5.2",
     "react-resizable": "^1.9.0",
+    "react-router-dom": "^5.1.2",
     "react-scripts": "3.2.0",
+    "react-social-login": "^3.4.6",
     "reactstrap": "^8.1.1"
   },
   "scripts": {
-    "start": "react-scripts start",
+    "start": "set HTTPS=true&&react-scripts start",
```
Comment (blazinaj) [inline package.json:27]: What does this line do? Does it force HTTPS for localhost?

File: src/index.js [hunk 2]
```diff
@@ -5,9 +5,22 @@ import 'react-grid-layout/css/styles.css';
...
+const routing = (
+    <Router>
+        <div>
+            <Switch>
+                <Route exact path="/" component={App} />
+                <Route path="/config" component={Configuration} />
+            </Switch>
+        </div>
+    </Router>
+);
```
Comment (blazinaj) [inline src/index.js:21]: Sweet! Thanks for adding routing

File: src/views/Config/Config.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        this.onLoginSuccess = this.onLoginSuccess.bind(this);
+        this.onLoginFailure = this.onLoginFailure.bind(this);
+        this.onLogoutSuccess = this.onLogoutSuccess.bind(this);
+        this.onLogoutFailure = this.onLogoutFailure.bind(this);
+        this.logout = this.logout.bind(this)
+    }
+
+    setNodeRef(provider, node) {
+        if (node) {
+            this.nodes[provider] = node
```
Comment (blazinaj) [inline src/views/Config/Config.js:27]: What is this doing?

**Review by blazinaj — APPROVED** [review]
Very cool. The coding style is a little inconsistent, but I'm guessing that's due to the examples. Thank you for stubbing out the social logins for the other providers as well, looks like it will be simple to implement those.

File: package.json (hunk 1 above)
Comment (toha87) [reply package.json:27]: yes, because some providers require this

File: src/views/Config/Config.js (hunk 3 above)
Comment (toha87) [reply src/views/Config/Config.js:27]: keeps the reference to Providers object in our case we have 3

**Comment by toha87** [comment]
> Very cool. The coding style is a little inconsistent, but I'm guessing that's due to the examples. Thank you for stubbing out the social logins for the other providers as well, looks like it will be simple to implement those.

Thanks for showing me the npm package. It worked out really well.

**Pull request merged**